src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -801,26 +876,52 @@ private void createAndSetStatistics()
                 case DELIVERYMEN_JOB:
                     deliverymen++;
                     break;
+                case MINER_JOB:
+                    miners++;
+                    break;
+                case FISHERMAN_JOB:
+                    fishermen++;
+                    break;
+                case LUMBERJACK_JOB:
+                    lumberjacks++;
+                    break;
+                case FARMER_JOB:
+                    farmers++;
+                    break;
+                case GUARD_JOB:
+                    guards++;
+                    break;
                 case "":
                     break;
                 default:
                     workers++;
             }
         }
 
-        workers += deliverymen + builders;
+        workers += deliverymen + builders + miners + fishermen + lumberjacks + farmers + guards;
 
         final String numberOfCitizens =
           LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.totalCitizens", citizensSize, townHall.getColony().getMaxCitizens());
         final String numberOfUnemployed = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.unemployed", citizensSize - workers);
         final String numberOfBuilders = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.builders", builders);
         final String numberOfDeliverymen = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.deliverymen", deliverymen);
-
-        findPaneOfTypeByID(HAPPYNESS_LABEL, Label.class).setLabelText(Double.toString(building.getColony().getOverallHappiness()));
+        final String numberOfMiners = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.miners", miners);
+        final String numberOfFishermen = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.fishermen", fishermen);
+        final String numberOfGuards = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.Guards", guards);
+        final String numberOfLumberjacks = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.lumberjacks", lumberjacks);
+        final String numberOfFarmers = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.farmers", farmers);
+        final int roundedHappiness = (int) Math.round(building.getColony().getOverallHappiness());
+
+        findPaneOfTypeByID(HAPPINESS_LABEL, Label.class).setLabelText(Integer.toString(roundedHappiness));
```

Comment:
I think rounding the happiness to an integer might be a bit too much.
I think rounding it to 1 digit after the decimal point is better, it shows the player the increase without too much detail.